### PR TITLE
[graphql/rpc] use db over rpc 1.0 in tests

### PR DESCRIPTION
**Pull request by oxade**
## Description 

Refactors some things so we uss DB over RPC 1.0 in tests

## Test Plan 

Existing

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/src/server/builder.rs [hunk 1]
```diff
@@ -198,15 +198,12 @@ async fn graphiql() -> impl axum::response::IntoResponse {
     )
 }
 
-#[cfg(test)]
```
Comment (wlmyng) [inline crates/sui-graphql-rpc/src/server/builder.rs:201]: shouldnt we still keep this flag?

File: crates/sui-graphql-rpc/tests/e2e_tests.rs [hunk 2]
```diff
@@ -130,4 +130,30 @@ mod tests {
             .await
             .unwrap();
     }
+
+    use sui_graphql_rpc::server::builder::tests::*;
+
+    #[tokio::test]
+    #[serial]
+    async fn test_timeout() {
+        test_timeout_impl().await;
```
[the remark this answers is not in the document]
Comment (oxade) [reply crates/sui-graphql-rpc/tests/e2e_tests.rs:139]: The tests touch private members so its easier to construct them locally and jus expose the functions.
The serial macro makes sure tests are not run in parallel

File: crates/sui-graphql-rpc/src/server/builder.rs (hunk 1 above)
Comment (oxade) [reply crates/sui-graphql-rpc/src/server/builder.rs:201]: Tried but but then they wont be visible to the integration tests.

**Review by wlmyng — APPROVED** [review]

File: crates/sui-graphql-rpc/Cargo.toml [hunk 3]
```diff
@@ -28,6 +28,7 @@ prometheus.workspace = true
 rand.workspace = true # todo: cleanup test only deps
 regex.workspace = true
 reqwest.workspace = true
+serial_test = { version = "2.0.0"}
```
Comment (amnn) [inline crates/sui-graphql-rpc/Cargo.toml:31]: @oxade, is there a reason this isn't a `workspace` dependency? It's the only one.

File: crates/sui-graphql-rpc/src/server/builder.rs (hunk 1 above)
Comment (amnn) [reply crates/sui-graphql-rpc/src/server/builder.rs:201]: We should move this into a test helper crate in that case -- we shouldn't leak test helpers into the default build configuration.

File: crates/sui-graphql-rpc/Cargo.toml (hunk 3 above)
Comment (oxade) [reply crates/sui-graphql-rpc/Cargo.toml:31]: It was just convenient to plop it in, but sure addressed here https://github.com/MystenLabs/sui/pull/14489

**Pull request merged**